host/ProtocolGateway.Host.Common/Bootstrapper.cs

```diff
@@ -53,16 +55,14 @@ public Bootstrapper(ISettingsProvider settingsProvider, ISessionStatePersistence
 
             this.settingsProvider = settingsProvider;
             this.settings = new Settings(this.settingsProvider);
+            this.iotHubClientSettings = new IotHubClientSettings(this.settingsProvider);
             this.sessionStateManager = sessionStateManager;
             this.qos2StateProvider = qos2StateProvider;
             this.authProvider = new SasTokenDeviceIdentityProvider();
-            this.iotHubMessageRouter = new ConfigurableMessageRouter();
+            this.topicNameConverter = new ConfigurableMessageAddressConverter();
```

Comment:
erm.. without it "I" is the only thing separating it from interface. Should have at least some specialization. Other ideas?